src/governance.cpp

```diff
@@ -211,13 +211,27 @@ void CGovernanceManager::ProcessMessage(CNode* pfrom, std::string& strCommand, C
 
         if(!fIsValid) {
             if(fMasternodeMissing) {
-                mapMasternodeOrphanObjects.insert(std::make_pair(nHash, object_time_pair_t(govobj, GetAdjustedTime() + GOVERNANCE_ORPHAN_EXPIRATION_TIME)));
+
+                int& count = mapMasternodeOrphanCounter[govobj.GetMasternodeVin().prevout];
+                if (count >= 10) {
+                    LogPrint("gobject", "MNGOVERNANCEOBJECT -- Too many orphan objects, missing masternode=%s\n", govobj.GetMasternodeVin().prevout.ToStringShort());
+                    // ask for this object again in 2 minutes
```

Comment:
Yes, `mapAskFor` is processed by `SendMessages`, but request (`PushMessage`) can not be sent earlier than key value (`nRequestTime` in `CNode::AskFor`)

Actually, `AskFor` is intended to ensure that we are requesting an object just once (avoid multiple non-responded requests) and not too often (each retry is 2 minutes after the last).